src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/model/entity/session/PermittedSessionLocation.kt

```diff
@@ -0,0 +1,56 @@
+package uk.gov.justice.digital.hmpps.visitscheduler.model.entity.session
+
+import org.hibernate.annotations.CreationTimestamp
+import org.hibernate.annotations.UpdateTimestamp
+import org.springframework.data.jpa.repository.Temporal
+import uk.gov.justice.digital.hmpps.visitscheduler.model.entity.Prison
+import java.time.LocalDateTime
+import javax.persistence.Column
+import javax.persistence.Entity
+import javax.persistence.GeneratedValue
+import javax.persistence.GenerationType
+import javax.persistence.Id
+import javax.persistence.JoinColumn
+import javax.persistence.ManyToMany
+import javax.persistence.ManyToOne
+import javax.persistence.Table
+import javax.persistence.TemporalType
+
+@Entity
+@Table(name = "PERMITTED_SESSION_LOCATION")
+data class PermittedSessionLocation(
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  @Column(name = "ID")
+  val id: Long = 0,
+
+  @Column(name = "PRISON_ID", nullable = false)
+  val prisonId: Long,
+
+  @ManyToOne
+  @JoinColumn(name = "PRISON_ID", updatable = false, insertable = false)
+  val prison: Prison,
+
+  @ManyToMany(mappedBy = "permittedSessionLocations")
+  var sessionTemplates: MutableList<SessionTemplate> = mutableListOf(),
```

Comment:
we could make this var private so that the getter is not exposed but am not sure if it impacts how JPA works.

Reply:
Not a good idea with JPA objects as they are wrapped by proxy classes and when you create a new Session Template you would have to add it to an existing location object after it has been created by the JPA frame work so you would need to have access to sessionTemplates attribute to add after creation.